checked_yaml/lib/checked_yaml.dart

```diff
@@ -27,8 +27,17 @@ T checkedYamlDecode<T>(
   allowNull ??= false;
   YamlNode yaml;
 
+  Uri sourceUri;
+  if (sourceUrl == null) {
+    // noop
+  } else if (sourceUrl is Uri) {
+    sourceUri = sourceUrl;
+  } else {
+    sourceUri = Uri.parse(sourceUrl as String);
+  }
+
   try {
-    yaml = loadYamlNode(yamlContent, sourceUrl: sourceUrl);
+    yaml = loadYamlNode(yamlContent, sourceUrl: sourceUri);
```

Comment:
I would recommend pushing `Uri` up to callers.

Reply:
will do on the next breaking changed – just not now